docs/_static/js/search.js

```diff
@@ -0,0 +1,191 @@
+/* eslint no-use-before-define: 0 */
+/* eslint @typescript-eslint/no-use-before-define: 0 */
+/* eslint no-underscore-dangle: 0 */
+/* eslint no-undef: 0 */
+/* eslint no-restricted-syntax: 0 */
+/* eslint prefer-destructuring: 0 */
+/* eslint no-console: 0 */
```

Comment:
it would be best to resolve these linting errors a different way.

* First, try to refactor the JS to avoid them (no-use-before-define) should be easy, same with no-underscore-dangle and no-undef
* for others, try to put the eslint ignore in the actual line of the change (e.g. for the console, I think there is only one line)
* This helps us ensure that new code added to this file adheres to the eslint styleguide
* I know you copied/pasted this code and it simply was not linted before, but we should try to have a balance of fixing things that are important for code readability/bugs when bring to its own file.

Reply:
@lb- , refactored JavaScript files to remove most of the linting errors. I even removed inline JS code from layout.html and merged it with search.js.
Added //eslint-disable to only unavoidable specific Lines of Code.